From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 23 Feb 2022 18:11:22 -0500
Subject: [PATCH] Customize Windows NSIS installer script.

1. Never clear the full pkg dir
2. Delete environment variables set in registry by PothosSDR
3. Do not show "advanced" installation options
---
 constructor/nsis/main.nsi.tmpl | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 2426248..a7b6f1d 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -118,7 +118,7 @@ Page Custom InstModePage_Create InstModePage_Leave
 !define MUI_PAGE_CUSTOMFUNCTION_LEAVE OnDirectoryLeave
 !insertmacro MUI_PAGE_DIRECTORY
 # Custom options now differ depending on installation mode.
-Page Custom mui_AnaCustomOptions_Show
+#Page Custom mui_AnaCustomOptions_Show
 !insertmacro MUI_PAGE_INSTFILES
 !insertmacro MUI_PAGE_FINISH
 
@@ -933,7 +933,7 @@ Section "Install"
 
     ${If} $Ana_ClearPkgCache_State = ${BST_CHECKED}
         DetailPrint "Clearing package cache..."
-        push '"$INSTDIR\_conda.exe" clean --all --force-pkgs-dirs --yes'
+        push '"$INSTDIR\_conda.exe" clean --all --yes'
         push 'Failed to clear package cache'
         call AbortRetryNSExecWait
     ${EndIf}
@@ -964,6 +964,13 @@ Section "Install"
     ${EndIf}
 
     ${If} $ARGV_NoRegistry == "0"
+        # Delete registry entries for environment variables set by PothosSDR
+        DeleteRegValue HKEY_LOCAL_MACHINE "SYSTEM\CurrentControlSet\Control\Session Manager\Environment" "GR_PREFIX"
+        DeleteRegValue HKEY_LOCAL_MACHINE "SYSTEM\CurrentControlSet\Control\Session Manager\Environment" "GRC_BLOCKS_PATH"
+        DeleteRegValue HKEY_LOCAL_MACHINE "SYSTEM\CurrentControlSet\Control\Session Manager\Environment" "UHD_PKG_PATH"
+        DeleteRegValue HKEY_LOCAL_MACHINE "SYSTEM\CurrentControlSet\Control\Session Manager\Environment" "VOLK_PREFIX"
+        #ClearErrors
+
         # Registry uninstall info
         WriteRegStr SHCTX "${UNINSTREG}" "DisplayName" "${UNINSTALL_NAME}"
         WriteRegStr SHCTX "${UNINSTREG}" "DisplayVersion" "${VERSION}"
-- 
2.35.0

